### PR TITLE
Social sharing: Tile & button to encourage in-messenger sharing of Oscars Bot Game

### PR DESCRIPTION
**Pull request by annebyrne**

### CODE REVIEW

**Comment by tomrf1** [comment]
Broken tests :(

File: src/main/scala/com/gu/facebook_news_bot/state/OscarsNomsStates.scala [hunk 1]
```diff
@@ -188,18 +187,86 @@ object OscarsNomsStates {
...
     }
 
   }
 
+  case object UpdateTypeState extends State {
+
+    val Name = "OSCARS_UPDATE_TYPE_QUESTION"
+
+    def transition(user: User, messaging: MessageFromFacebook.Messaging, capi: Capi, facebook: Facebook, store: UserStore): Future[Result] = {
+      println(Name)
```
Comment (tomrf1) [inline src/main/scala/com/gu/facebook_news_bot/state/OscarsNomsStates.scala]: debugging?

File: src/main/scala/com/gu/facebook_news_bot/state/OscarsNomsStates.scala [hunk 2]
```diff
@@ -188,18 +187,86 @@ object OscarsNomsStates {
...
+
+      State.getUserInput(messaging).flatMap { text =>
+        val lower = text.toLowerCase
+        println(s"[1] : ${lower}")
```
Comment (tomrf1) [inline src/main/scala/com/gu/facebook_news_bot/state/OscarsNomsStates.scala]: ditto

File: src/main/scala/com/gu/facebook_news_bot/state/OscarsNomsStates.scala [hunk 3]
```diff
@@ -188,18 +187,86 @@ object OscarsNomsStates {
...
+
+      val quickReplies = Seq(
+        MessageToFacebook.QuickReply(title = Some("Rolling updates"), payload = Some("rolling-updates")),
+        MessageToFacebook.QuickReply(title = Some("Morning briefing"), payload = Some("morning-briefing"))
+      )
+
+      val message = MessageToFacebook.quickRepliesMessage(
+        user.ID,
+        quickReplies,
+        "Thanks for your votes. We’ll be in touch with stats, news and results. When would you like your Oscar winner updates?"
```
Comment (tomrf1) [inline src/main/scala/com/gu/facebook_news_bot/state/OscarsNomsStates.scala]: Is this clear enough? Is it worth explaining what the 2 options mean? Will people realise we're going to message them through the night?

File: src/main/scala/com/gu/facebook_news_bot/state/OscarsNomsStates.scala [hunk 4]
```diff
@@ -188,18 +187,86 @@ object OscarsNomsStates {
...
+      )
+
+      (State.changeState(user, Name), List(message))
+    }
+
+    def updateUserUpdateType(user: User, rollingUpdates: Boolean): Future[Result] = {
+      val updatedUser = user.copy(
+        state = Some(UpdateTypeState.Name),
```
Comment (tomrf1) [inline src/main/scala/com/gu/facebook_news_bot/state/OscarsNomsStates.scala]: unnecessary - we're about to put them into the MAIN state

File: src/main/scala/com/gu/facebook_news_bot/state/OscarsNomsStates.scala [hunk 5]
```diff
@@ -188,18 +187,86 @@ object OscarsNomsStates {
...
+      )
+
+      triggerSocialShare(updatedUser)
+      val message = List(
+        MessageToFacebook.textMessage(user.ID, "Great, thanks for playing! Is there anything else I can help you with?")
+      )
+      Future.successful(State.changeState(user, MainState.Name), message)
+    }
+
+    def triggerSocialShare(user: User): Future[Result] = {
```
Comment (tomrf1) [inline src/main/scala/com/gu/facebook_news_bot/state/OscarsNomsStates.scala]: Can we make this something like `def getSocialShareElement(user: User): MessageToFacebook.Element = ...` and use the result to build the message?

**Comment by annebyrne** [comment]
tests should now be fixed @tomrf1  !

**Comment by tomrf1** [comment]
👍

**Pull request merged**